samples/hello++/main.cpp

```diff
@@ -0,0 +1,40 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include "stdio.h"
+
+class hello{
+private:
+  int hiddenValue = 0;
+public:
+  hello() {hiddenValue = 5;};
+  hello(int v) : hiddenValue(v) {};
+  int getValue() { return hiddenValue; };
+};
+
+int main() {
+  switch(pb_init())
+  {
+  case 0: break;
+  default:
+    XSleep(2000);
+    XReboot();
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  hello h(6);
+  hello i;
+  
+  while(1) {
+    debugPrint("Value of h: %i\n", h.getValue());
+    debugPrint("Value of i: %i\n", i.getValue());
+    XSleep(2000);
+  }
+
+  pb_kill();
+  XReboot();
```

Comment:
I believe nxdk reboots automatically when returning from `main`. I consider this the equivalent of `exit(0); return 0;`.

We should fix this in all of our samples.

Reply:
I tested it in XQEMU, returning simply restarts the program, `XReboot()` crashes with something nv2a related. I'd argue they're not interchangeable.
Alas, I can't test this on real hardware currently.

Reply:
We should fix this then. If a program returns from main, that should behave like an "exit" (including `atexit()` callbacks / destructors) and I'd argue that the strategy on Xbox to "exit" is to:

- Reboot to dashboard
- Reboot to the next program (Dashboard, or DVD if one is inserted)

I'm not sure which of these XReboot does, but it's safe to assume it should be one of them.
Eitherway, we shouldn't need XReboot in our sample code.

Reply:
Agreed, this should be looked into further, but since it affects all samples perhaps this would be better served as a separate issue?